### PR TITLE
Refactor to split interface/transport

### LINKED ISSUE #31: Adding back in Support for TCP Connections
Hello, the code is great! Would it be possible to add back in the support for TCP connections as previously?
https://github.com/numat/alicat/tree/8af92647cb396401c0d604d83fb95a49d9a82be9

It's incredibly useful!

### LINKED ISSUE #35: Incorrect readings from locked devices
```
❯ alicat /dev/tty.usbserial-FTCJ5EK9 --lock
{
  "control_point": null,
  "gas": "LCK",
  "mass_flow": "+000.00",
  "pressure": "+014.46",
  "setpoint": "000.00",
  "temperature": "+026.54",
  "total flow": "Air",
  "volumetric_flow": "+000.00"
}
```

### LINKED ISSUE #36: Numeric values are incorrectly returned as strings
https://github.com/numat/alicat/commit/6e6c1ad4599582c7bacf1a7a0c51b1825c0bc1bd was overzealous, and returns numeric values as strings.

```
❯ alicat /dev/tty.usbserial-FTCJ5EK9
{
  "control_point": "pressure,
  "gas": "Air",
  "mass_flow": "-000.00",
  "pressure": "+014.60",
  "setpoint": 0.05,
  "temperature": "+028.24",
  "volumetric_flow": "-000.00"
}
```

### PR DESCRIPTION
**Pull request by alexrudd2**
- Switch fully to `asyncio`.
- Setup linting and testing.
- Split interface and transport layers
- Closes #31
- Closes #35
- Closes #36

### CODE REVIEW

File: alicat/driver.py [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+         * Total flow (only on models with the optional totalizer function)
+         * Currently selected gas
+
+        Args:
+            retries: Number of times to re-attempt reading. Default 2.
+        Returns:
+            The state of the flow controller, as a dictionary.
+
+        """
+        command = f'{self.unit}'
```
Comment (patrickfuller) [inline alicat/driver.py:128]: This line's not doing anything, right?

File: alicat/driver.py [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+        if not self.open:
+            raise OSError(f"The FlowMeter with unit ID {self.unit} and "
+                           "port {self.hw.address} is not open")
+
+    def _is_float(self, msg):
+        try:
+            float(msg)
+            return True
+        except ValueError:
+            return False
```
Comment (patrickfuller) [inline alicat/driver.py:103]: Pedantic, but I'd move outside of the class.

File: alicat/driver.py [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+        total_percent = sum(gases.values())
+        if total_percent != 100:
+            raise ValueError("Percentages of gas mix must add to 100%!")
+
+        if any(gas not in self.gases for gas in gases):
+            raise ValueError("Gas not supported!")
+
+        gas_list = ' '.join(
+            [' '.join([str(percent), str(self.gases.index(gas))])
+                for gas, percent in gases.items()])
```
Comment (patrickfuller) [inline alicat/driver.py:210]: Not very readable. Maybe...

```python

gas_list = [f'{percent} {self.gases.index(gas)}' for gas, percent in gases.items()]
command = ' '.join([self.unit,
                    'GM',
                    name,
                    str(mix_no),
                    ' '.join(gas_list)])
```

File: alicat/driver.py (hunk 1 above)
Comment (alexrudd2) [reply alicat/driver.py:128]: Screenshot 2023-04-12 at 12 30 22 PM

File: alicat/util.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                               'stopbits': stopbits,
+                               'parity': parity,
+                               'timeout': timeout}
+        self.ser = serial.Serial(self.address, **self.serial_details)
+
+    async def _read(self, length: int):
+        """Read a fixed number of bytes from the device."""
+        return self.ser.read(length).decode()
+
+    async def _readline(self):
```
Comment (patrickfuller) [inline alicat/util.py:211]: I think the higher-level `write_and_read` will block with this serial implementation. There needs to be an async wait-until-ready part.

I don't know how best to do this with serial today.

**Comment by patrickfuller** [comment]
Quick review. Only comment that matters is that I think the serial comm is still blocking. Don't know what the pattern to fix that today is, but it does fell weird for the end user to `await` commands that aren't async.

All other comments are pedantic one-offs and can be ignored.

File: alicat/driver.py (hunk 2 above)
Comment (alexrudd2) [reply alicat/driver.py:103]: 1fc791397e226cac6dfa50770fe833c9d232c0fa

File: alicat/driver.py (hunk 3 above)
Comment (alexrudd2) [reply alicat/driver.py]: b6a37900b07e3e5434bb88be8cc7788ff1fce1e3

Comment (alexrudd2) [reply alicat/driver.py]: Reminder to me: test again on real hardware

**Pull request merged**